src/realm/bplustree.hpp

```diff
@@ -458,6 +458,20 @@ class BPlusTree : public BPlusTreeBase {
             set(ndx1, tmp2.get());
             set(ndx2, tmp1.get());
         }
+        else if constexpr (std::is_same_v<T, Mixed>) {
+            std::string buf1;
+            std::string buf2;
+            Mixed tmp1 = get(ndx1);
+            Mixed tmp2 = get(ndx2);
+            if (tmp1.is_type(type_String, type_Binary)) {
+                tmp1.use_buffer(buf1);
```

Comment:
Can we use the same concept as the SwapBuffer above instead of making copies of the data?

Reply:
oh, sorry I see we are not copying the data, this is fine with me

Reply:
Well - we are copying the data - as we do with SwapBuffer.

Reply:
I think I'm getting confused, but where is SwapBuffer making a copy? It appears to use the std::string constructor which takes a pointer and size, which AFAIK does not copy the data. Then at `get()` it uses the StringData constructor which also does not copy.

Reply:
When you create a std:string from StringData object, you copy the string. std::string owns the data as opposed to std::string_view.

Reply:
My bad, thanks for your patience!